src/components/generic-link-share/component.tsx

```diff
@@ -70,6 +76,60 @@ function GenericLinkShare(
     }
   };
 
+  const requestCurrentPage = (currentTxtUri: string) => fetch(currentTxtUri)
+    .then((response) => response.text());
+
+  const addButtonToPresentationToolbar = (title: string, viewer: string, presenter?: string) => {
+    if (currentUser?.presenter) {
+      const currentObjectToSendToClient = new PresentationToolbarButton({
+        label: `Play ${title}`,
+        tooltip: 'A generic link tag has been detected in this slide, show it to all?',
+        onClick: () => {
+          deleteEntryUrlToGenericLink([RESET_DATA_CHANNEL]);
+          pushEntryUrlToGenericLink({
+            url: presenter || viewer,
+            isUrlSameForRole: !presenter,
+            viewerUrl: viewer,
+          });
+        },
+      });
+      pluginApi.setPresentationToolbarItems([currentObjectToSendToClient]);
+    }
+  };
+
+  const handleFetchPresentationData = (
+    currentPres: CurrentPresentation,
+  ) => {
+    pluginApi.setPresentationToolbarItems([]);
+    const currentTxtUri = currentPres?.currentPage?.urlsJson?.text;
+    pluginLogger.info('Trying to fetch the slide text in the following URI:', currentTxtUri);
+    if (currentTxtUri) {
+      requestCurrentPage(currentTxtUri).then((currentPageContent) => {
+        const match = SEARCH_PATTERN.exec(currentPageContent);
+        if (match && match.length > 0) {
+          pluginLogger.info('Link tag found! Parsing its fields...');
+          const linkContent = match[0];
+          const regexFields = /link\s*\{(?:[^{}]*\n)*?[^{}]*title:\s*“([^”]*)”(?:[^{}]*\n)*?(?:,\s*presenter:\s*“([^”]*)”(?:[^{}]*\n)*?)?(?:,\s*viewer:\s*“([^”]*)”(?:[^{}]*\n)*?)?[^{}]*\}/g;
```

Comment:
Please adjust the regex pattern to at least include quoted properties from a json string. it needs to identify 
```json
{
  "property1": "value1"
}
```
as a valid json.